src/io/flutter/project/FlutterIconProvider.java

```diff
@@ -20,7 +21,7 @@
 public class FlutterIconProvider extends IconProvider {
   @Nullable
   public Icon getIcon(@NotNull final PsiElement element, @Iconable.IconFlags final int flags) {
-    if (element instanceof PsiDirectory) {
+    if (element instanceof PsiDirectory && FlutterModuleUtils.hasFlutterModule(element.getProject())) {
```

Comment:
Is it possible to detect flutter module just by looking files / folders around?
`hasFlutterModule()` looks too heavy, it even may result in reading a file (`declaresFlutterDependency()`).

BTW is `if (!PlatformUtils.isIntelliJ()) ` branch needed in `hasFlutterModule()`? I thought `FlutterModuleType` used in small IDEs as well. (If it's not the case it means that you can't work with the same project in different IDEs. I hope it;s not the case).